src/pymovements/utils/parsing.py

```diff
@@ -253,10 +298,46 @@ def parse_eyelink(
         schema_overrides=schema_overrides,
     )
 
-    # convert the defaultdict to a regular dict
-    return_metadata = dict(metadata)
+    return df, pre_processed_metadata
+
+
+def _pre_process_metadata(metadata: defaultdict[str, Any]) -> dict[str, Any]:
+    """Pre-process metadata to suitable types and formats.
+
+    Parameters
+    ----------
+    metadata: defaultdict[str, Any]
+        Metadata to pre-process.
+
+    Returns
+    -------
+    dict[str, Any]
+        Pre-processed metadata.
+    """
+    # in case the version strings have not been found, they will be empty strings (defaultdict)
+    metadata['version_number'], metadata['model'] = _parse_full_eyelink_version(
+        metadata['version_1'], metadata['version_2'],
+    )
+
+    if 'resolution' in metadata:
+        resolution = metadata['resolution'].split()
+        metadata['resolution'] = tuple(int(res) for res in resolution)
+
+    if 'sampling_rate' in metadata:
+        metadata['sampling_rate'] = float(metadata['sampling_rate'])
+
+    if 'day' in metadata:
```

Comment:
can you also create a full [datetime](https://docs.python.org/3/library/datetime.html#datetime-objects) object here and save it to `metadata['datetime']`?